plutus-core/plutus-core/src/PlutusCore/ParserCommon.hs

```diff
@@ -69,64 +54,65 @@ intern n = do
             put $ ParserState identifiers'
             return fresh
 
-type Parser = ParsecT ParseError T.Text (StateT ParserState PLC.Quote)
-instance (Stream s, PLC.MonadQuote m) => PLC.MonadQuote (ParsecT e s m)
-
-parse :: Parser a -> String -> T.Text -> Either (ParseErrorBundle T.Text ParseError) a
+parse :: Parser SourcePos a -> String -> T.Text -> Either (ParseErrorBundle T.Text (PLC.ParseError SourcePos)) a
 parse p file str = PLC.runQuote $ parseQuoted p file str
 
-parseQuoted :: Parser a -> String -> T.Text -> PLC.Quote (Either (ParseErrorBundle T.Text ParseError) a)
+parseQuoted :: Parser SourcePos a -> String -> T.Text -> PLC.Quote
+                   (Either (ParseErrorBundle T.Text (PLC.ParseError SourcePos)) a)
 parseQuoted p file str = flip evalStateT initial $ runParserT p file str
 
-whitespace :: Parser ()
+whitespace :: Parser SourcePos ()
```

Comment:
Are we ever going to use something that is not `SourcePos`? Perhaps worth having a note explaining what we need the generality for.

Reply:
I'm not completely sure at this point. It will become clearly when I've cleaned up the other parts of the parser. I can add a note about that.